chaincode/compute_plan.go

```diff
@@ -311,13 +311,62 @@ func getComputePlan(db *LedgerDB, key string) (resp outputComputePlan, err error
 		}
 	}
 
+	status, _ := getComputePlanStatus(db, aggregatetupleKeys, traintupleKeys, compositeTraintupleKeys)
+
 	resp = outputComputePlan{
 		ComputePlanID:           key,
 		ObjectiveKey:            objectiveKey,
 		TraintupleKeys:          traintupleKeys,
 		CompositeTraintupleKeys: compositeTraintupleKeys,
 		AggregatetupleKeys:      aggregatetupleKeys,
 		TesttupleKeys:           testtupleKeys,
+		Status:                  status,
 	}
 	return
 }
+
+func getComputePlanStatus(db *LedgerDB, aggregatetupleKeys, traintupleKeys, compositeTraintupleKeys []string) (status string, err error) {
+	// get all tuples like status in one slice
+	statusCollection := []string{}
+
+	for _, aggregatetupleKey := range aggregatetupleKeys {
+		aggregatetuple, err := getOutputAggregatetuple(db, aggregatetupleKey)
```

Comment:
Why getting the output tuples here? If there is no specific need you should only get the intern object (here it will be `db.GetAggregatetuple`). More over it exists a function `db.GetGenericTuple` that work for every tuple and allow you to acces common part of tuples including the status.

Reply:
fixed here ef97caf